agent/src/com/cloud/agent/Agent.java

```diff
@@ -166,7 +178,8 @@ public Agent(final IAgentShell shell, final int localAgentId, final ServerResour
             throw new ConfigurationException("Unable to configure " + _resource.getName());
         }
 
-        _connection = new NioClient("Agent", _shell.getHost(), _shell.getPort(), _shell.getWorkers(), this);
+        final String host = _shell.getHost();
```

Comment:
@DaanHoogland yes, the `getHost` is supposed to internally loop/return the next address.